cscs-checks/apps/tensorflow/tensorflow_check.py

```diff
@@ -18,38 +14,43 @@ def __init__(self, model_name, **kwargs):
         self.tags = {'production'}
         self.num_tasks = 1
         self.num_gpus_per_node = 1
-        self.modules = ['TensorFlow/1.4.1-CrayGNU-17.08-cuda-8.0-python3']
+        self.modules = ['TensorFlow/1.7.0-CrayGNU-18.07-cuda-9.1-python3']
 
         # Checkout to the branch corresponding to the module version of
         # TensorFlow
-        self.pre_run = ['git checkout r1.4.0']
+        self.pre_run = ['git checkout r1.7.0']
+        self.variables = {'PYTHONPATH': '$PYTHONPATH:.'}
 
 
+@rfm.simple_test
 class TensorFlowMnistTest(TensorFlowBaseTest):
-    def __init__(self, **kwargs):
-        super().__init__('mnist', **kwargs)
+    def __init__(self):
+        super().__init__('mnist')
+
+        train_epochs = 10
         self.executable = 'python3 ./official/mnist/mnist.py'
         self.executable_opts = ['--model_dir', '.', '--export_dir', '.',
-                                ' --data_dir', '.']
+                                ' --data_dir', '.', '--train_epochs',
+                                str(train_epochs)]
 
         self.sanity_patterns = sn.all([
             sn.assert_found(r'INFO:tensorflow:Finished evaluation at',
                             self.stderr),
             sn.assert_gt(sn.extractsingle(
                 r"Evaluation results:\s+\{.*'accuracy':\s+(?P<accuracy>\S+)"
-                r"(?:,|\})", self.stdout, 'accuracy', float), 0.99)
+                r"(?:,|\})", self.stdout, 'accuracy', float, -1), 0.99)
```

Comment:
What was the difference between the first accuracy and the last accuracy value. I mean which one is more correct?

Reply:
Now I need to take the final value because there are many accuracy numbers printed throughout the calculation.